apps/src/code-studio/progressRedux.js

```diff
@@ -636,48 +635,50 @@ export function statusForLevel(level, levelProgress) {
 }
 
 /**
- * Groups lessons (aka stages) according to category.
+ * Groups lessons according to LessonGroup.
  * @returns {Object[]}
  * {string} Object.name
  * {string[]} Object.lessonNames
  * {Object[]} Object.stageLevels
  */
-export const categorizedLessons = (state, includeBonusLevels = false) => {
-  let byCategory = {};
+export const groupedLessons = (state, includeBonusLevels = false) => {
+  let byGroup = {};
 
   const allLevels = levelsByLesson(state);
 
-  state.stages.forEach((stage, index) => {
-    const category = stage.flex_category;
-    const lesson = lessonFromStageAtIndex(state, index);
-    let stageLevels = allLevels[index];
+  state.stages.forEach((lesson, index) => {
+    const group = lesson.lesson_group_display_name
```

Comment:
This is the key change to this file. Everything else is just renaming. This allows us to live in a world where we can use flex category or lesson group until we have moved completely over to lesson group

Reply:
@dmcavoy does this mean that if we found that we broke something, we would only revert the change to this line to go back to using flex categories?

Reply:
Yes

Reply:
now that everything will already have a lesson group, does using the conditional buy us anything there? I would suggest, just making this always use the lesson group display name. or, if necessary, you could make it conditional upon an experiment first, so that we could verify in production that everything still looks ok using the experiment flag, and then "launch" the experiment as a separate step.